Lint and Test job fails: missing architect binary and duplicate clean target in Makefile

## Problem

The `Lint and Test` GitHub Actions job has been failing on PRs (e.g. Align files PR #20) with two distinct Makefile errors:

```
make: architect: No such file or directory
Makefile.gen.go.mk:30: warning: overriding recipe for target 'build'
Makefile.dev.mk:5: warning: ignoring old recipe for target 'build'
Makefile.gen.go.mk:89: *** target file 'clean' has both : and :: entries.  Stop.
```

This is unrelated to the dependency or alignment payload of the PRs that surface it -- the failure is reproducible because:

1. `make check` invokes the `architect` binary, which is not installed in the CI runner.
2. `Makefile.gen.go.mk` (auto-generated by `architect`) and one of the dev Makefiles both declare `clean` (with `:` and `::`), which `make` rejects.

The chronic failure prevents Renovate, architectbot Align files, and any other automated PRs from auto-merging in this repo.

## Proposed solution

- Either install `architect` in the `Lint and Test` workflow before running `make check`, or replace the `make check` invocation with the canonical Go test/lint commands the workflow already runs elsewhere.
- Reconcile the duplicate `clean` target so only one declaration remains (likely drop the entry from `Makefile.gen.go.mk` and keep the one in the dev makefile, or vice versa).
- Re-run CI on Align files PR #20 to confirm green status, then auto-merge.

## Acceptance criteria

- The `Lint and Test` job passes on `main` and on architectbot Align files PRs.
- PR #20 (or its successor) auto-merges via `devctl pr approve-align-files` without manual intervention.

## References

- Failing run: https://github.com/giantswarm/k8senv/actions/runs/24662014862/job/72110149665
- klaus-lab trial: trials/2026-04-20-1430-marge-sweep.md

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -35,8 +35,11 @@ jobs:
           curl -fsSL "https://dl.k8s.io/v1.35.2/bin/linux/amd64/kube-apiserver" -o "$(go env GOPATH)/bin/kube-apiserver"
           chmod +x "$(go env GOPATH)/bin/kube-apiserver"
 
-      - name: Run Checks (Lint & Test)
-        run: make check
+      - name: Run lint
+        run: golangci-lint run --config .golangci.yml
+
+      - name: Run integration tests
+        run: make test-integration
 
       - name: Run govulncheck (Security Scan)
         run: make govulncheck

--- a/Makefile
+++ b/Makefile
@@ -5,9 +5,6 @@ include Makefile.*.mk
 # default goal 001
 .DEFAULT_GOAL := help
 
-.PHONY: clean
-clean:: ## Remove build artifacts and temp directories
-
 .PHONY: help
 help: ## Show this help message
 	@awk '/^##@/ { printf "\n\033[0;32m%s\033[0m\n", substr($$0, 5) } \

--- a/Makefile.lib.mk
+++ b/Makefile.lib.mk
@@ -26,7 +26,3 @@ lint-fix: ## Run golangci-lint with auto-fix
 deps: ## Download and tidy dependencies
 	go mod download
 	go mod tidy
-
-.PHONY: clean
-clean::
-	go clean

--- a/Makefile.test.mk
+++ b/Makefile.test.mk
@@ -42,8 +42,3 @@ coverage: ## Generate test coverage report (requires kine and kube-apiserver)
 .PHONY: coverage-html
 coverage-html: coverage ## Generate and open coverage report in browser
 	go tool cover -html=$(COVERAGE_FILE)
-
-
-.PHONY: clean
-clean::
-	rm -f $(COVERAGE_FILE)

--- a/Makefile.tools.mk
+++ b/Makefile.tools.mk
@@ -39,3 +39,8 @@ download-capi-crds: ## Download CAPI core CRDs
 .PHONY: clean-crds
 clean-crds: ## Remove downloaded CRDs
 	rm -f crds/*.yaml
+
+.PHONY: clean
+clean: ## Remove build artifacts and temp directories
+	go clean
+	rm -f $(COVERAGE_FILE)

